### DISCUSSION TITLE
PreRelease overrides keeps adding ".0"

### DISCUSSION THREAD

**Discussion by iampluque** [adamralph/minver#796] [Q&A] [answered]
My question is about that guy in bold: 1.0.1-PR **.0** .1+test

Major: 1
Minor: 0
Patch: 1
Prerelease: PR
**Who the heck are you!?**: 0
Height: 1
metadata: test

Is there a reason why if I use the defaultPreRelease override it keep adding the `.0` at the end of my override, as opposed of what's written in the FAQ: https://github.com/adamralph/minver#can-i-change-the-default-pre-release-phase-from-alpha-to-something-else

I know the technical reason, it's hardcoded that way here: https://github.com/adamralph/minver/blob/38b456ccb69e908f771fb70c4fdc3baec790cb9d/MinVer.Lib/Version.cs#L19

But is there another reason?

**Comment by adamralph** [comment]
Hi, please read https://github.com/adamralph/minver#how-it-works — it's all explained there.

**Comment by iampluque** [comment]
> Default pre-release identifiers are added. The default pre-release phase is "alpha", but this can be customised.

I read it and it still not explain the "0" behavior ... it says `alpha`, not `alpha.0`. Can you be more specific?

**Comment by adamralph** [comment, marked as answer]
The `0` is the pre-release "number". That is, you don't want to be restricted to a single pre-release, since it's likely you will want a 1st pre-release (`1.0.1.PR.1`), a 2nd pre-release (`1.0.1.PR.2`) etc. The "0th" pre-release is not one that you will release. It's just a way of versioning the interim builds between an RTM and your next pre-release.

E.g. your commits might be versioned as follows:

- `0.0.0.PR.0` (root commit)
- `0.0.0.PR.0.1`
- `0.0.0.PR.0.2`
- `1.0.0.PR.1` (tagged and released)
- `1.0.0.PR.1.1`
- `1.0.0.PR.1.2`
- `1.0.0.PR.2` (tagged and released)
- `1.0.0.PR.2.1`
- `1.0.0.PR.2.2`
- `1.0.0` (tagged and released)
- `1.0.1.PR.0.1` _<-- this is the version you are currently seeing_
- `1.0.1.PR.0.2`
- `1.0.1.PR.1` (tagged and released)
- `1.0.1.PR.1.1`
- `1.0.1.PR.1.2`
- `1.0.1.PR.2` (tagged and released)
- `1.0.1.PR.2.1`
- `1.0.1.PR.2.2`
- `1.0.1` (tagged and released)
- ...

Side note: you may want to consider using something other than "PR" as your pre-release phase. It's easily confused with "pull request". When I first saw it, I thought that's what it meant.

**Comment by iampluque** [comment]
💯 agree on side note! :')

Oh I just understood by re-reading you how-to's that you consider `PR` (in this case) as a pre-release **phase**. That is the important part that I didn't understood on your way to implement the pre-release identifier.

As `1.0.0.PR.1 (tagged and released)` is a manual tag, I wasn't understanding why you wouldn't just append the height after the pre-release identifier (`PR` I though in this case) .... now it's all clear 👍 

Thanks! Happy holidays!

**Comment by adamralph** [comment]
Great, thanks! You too!

**Comment by lonix1** [comment]
> it says `alpha`, not `alpha.0`

I was also confused by this, and the docs do not explain the concept of "phase".

When I read the above explanation  - which due to the example format, is extremely helpful - it all made sense. It would be good for future readers if it were incorporated somehow into the docs.

Thanks for the excellent example.

**Reply by adamralph** [reply]
@lonix1 I'll see what I can do about making this clearer in the readme.

**Reply by adamralph** [reply]
After more thought, I believe it would simpler to remove the concept of "phase" altogether, and instead refer to the _default pre-release identifiers_ as being `alpha.0`. That would involve deprecating, and eventually removing, `MinVerDefaultPreReleasePhase` in favour of `MinVerDefaultPreReleaseIdentifiers`. I'm going to pursue this solution to see where it leads.

**Reply by adamralph** [reply]
That work can be tracked in https://github.com/adamralph/minver/pull/839

**Reply by rafntor** [reply]
Would it be possible to skip the ".0" ?
It can explicitly added in the `MinVerDefaultPreReleaseIdentifiers` so for those who need it is available - or is there something I am missing?
In my use-case I would prefer the specified identifier to be applied unmodified. (Using the CI build-id for prerelease)

![image](https://user-images.githubusercontent.com/44544090/213861847-5f2e0687-de37-47a7-bc9c-82167441c5a7.png)

**Reply by rafntor** [reply]
Forget it - I tried the beta and it does it just like that , so thanks ! &#128640;

**Reply by rafntor** [reply]
This may be a corner-case ; but I would expect the commandline-supplied prerelase to take precence , instead it keeps the tagged prerelease-name. Is that intentional? 
![image](https://user-images.githubusercontent.com/44544090/213924348-27ede278-255a-4875-8061-3c6ad5cfd447.png)

**Reply by adamralph** [reply]
@rafntor yes, it is intentional. Please see _[How it works](https://github.com/adamralph/minver#how-it-works)_.

**Reply by rafntor** [reply]
Yes I can see it matches the documentation. There is different handling if the preceding tag is an RTM-tag or Non-RTM-tag, both add height but only RTM-tags considers the supplied pre-release name. I would have liked if the argument applied equally to both, but I suppose it is a matter of choice ;-)

**Reply by adamralph** [reply]
> only RTM-tags considers the supplied pre-release name

That's why they are called the _default_ pre-release identifiers. They are the defaults, to be used when the last tag doesn't have any pre-release identifiers (i.e. it's an RTM tag).

If you want to publish a version of your software with MinVer, you should tag it. Then you have complete control over the version. I would advise using the versions with height for anything other than internal troubleshooting (e.g. bisecting to isolate bug introduction, etc.).
